lib/primitives/mtx.js

```diff
@@ -1881,6 +1890,132 @@ class CoinSelector {
     this.fee = Math.min(this.hardFee, CoinSelector.MAX_FEE);
     this.fund();
   }
+
+  /**
+   * Initiate selection using new selection algorithm.
+   */
+
+  async selectByValue() {
+    // add dummy output for change.
+    const change = new Output();
+
+    if (this.changeAddress) {
+      change.script.fromAddress(this.changeAddress);
+    } else {
+      // In case we don't have a change address,
+      // we use a fake p2pkh output to gauge size.
+      change.script.fromPubkeyhash(Buffer.allocUnsafe(20));
+    }
+
+    this.tx.outputs.push(change);
+
+    let target = this.outputValue;
+
+    // calculate fee for everything except inputs.
+    let size = await this.tx.estimateSize();
+    const fee = this.getFee(size);
+    // add fee to target value.
+    target += fee;
+
+    // begin selection.
+
+    // while target is greater than
+    // the largest coin we have, we
+    // will keep selecting the largest coin
+
+    this.index = this.coins.length - 1;
+    let coin = this.coins[this.index];
+    let value = Amount.value(coin.value);
```

Comment:
You use `Amount.value()` a lot in this branch but I don't understand why. `Amount` is a UI module for converting (for example) an integer of satoshis to a string or float of BTC for the user. `coin.value` here should be an integer of satoshis and that should always be how we compare values internally. If the `target` is a float like `12.1304234 BTC` then that should be converted to an integer of satoshis first.